src/vs/workbench/contrib/chat/browser/chatImagePaste.ts

```diff
@@ -9,18 +9,18 @@ import { IClipboardService } from '../../../../platform/clipboard/common/clipboa
 import { IChatRequestVariableEntry } from '../common/chatModel.js';
 import { ChatInputPart } from './chatInputPart.js';
 import { localize } from '../../../../nls.js';
-import { IConfigurationService } from '../../../../platform/configuration/common/configuration.js';
+import { IExtensionService, isProposedApiEnabled } from '../../../services/extensions/common/extensions.js';
 
 export class ChatImageDropAndPaste extends Disposable {
 
 	constructor(
 		private readonly inputPart: ChatInputPart,
 		@IClipboardService private readonly clipboardService: IClipboardService,
-		@IConfigurationService private readonly configurationService: IConfigurationService
+		@IExtensionService private readonly extensionService: IExtensionService
 	) {
 		super();
 		this._register(this.inputPart.inputEditor.onDidPaste((e) => {
-			if (this.configurationService.getValue<boolean>('chat.experimental.imageAttachments')) {
+			if (this.extensionService.extensions.some(ext => isProposedApiEnabled(ext, 'chatReferenceBinaryData'))) {
```

Comment:
This would enable me to send binary data to an extension that doesn't support `chatReferenceBinaryData`, as long as some _other_ extension does, right? Are you missing a check at the `extHost*.ts` layer to check that only extensions that support this proposal get that data?